### DISCUSSION TITLE
I study hard .. but github kill me ...

### DISCUSSION THREAD

**Discussion by choichanhyeok** [community/community#33267] [Other Feature Feedback, Questions, & Ideas] [answered]
I studied hard about the database today ..
I wrote down my studies in the md file and commit it, and GitHub gave the 500 erros.

when I went back, all my weritings disappeareed ..

i'm so sad !!!!

<img width="634" alt="8856786578657878" src="https://user-images.githubusercontent.com/68278903/190661732-fb908dfc-4938-47da-9bf2-0896dbe7da33.PNG">
<img width="1030" alt="5436456456" src="https://user-images.githubusercontent.com/68278903/190661736-60d6db2b-a1eb-4928-8068-d646ab84ba8b.PNG">

**Comment by melvincwng** [comment]
@choichanhyeok , to prevent such incidents from happening in the future, you can do the following:

1. Clone the repository to your desktop
2. Open your .md file in your IDE of choice
3. Make your changes in your IDE
4. Save the file
5. Push your changes to GitHub using git commands (i.e. git add -A, git commit -m, git push origin main)

This would prevent your code changes from disappearing even if GitHub encounters an outage, as your code is saved locally in your computer.

In general, I don't recommend editing your code files directly on the GitHub website due to the reasons above as seen in your screenshots.

Hope this helps and answers your query 😊

**Reply by choichanhyeok** [reply]
i love you my friend ༼ つ ◕_◕ ༽つ

**Comment by airtower-luna** [comment, marked as answer]
I agree with @melvincwng that keeping local copies of your repositories is good. It also makes working faster.

But for your current problem: Error 500 means the server has some kind of problem. If the issue doesn't go away soon, you should [send a bug report](https://support.github.com/contact/bug-report) to GitHub support if you haven't already. That way GitHub engineers will know about your problem and can work on fixing it.

**Reply by choichanhyeok** [reply]
I thought GitHub was a god because I was a beginner developer. It turns out you're the gods

(❁´◡`❁)
